test/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,675 @@
+
+  A1 = R[x y; x^2 y^2]
+  A2 = R[x+x^2 y+y^2]
+  @test Oscar.SubQuo(F2, A2, B1) == intersect(Oscar.SubQuo(F2, A1,B1), Oscar.SubQuo(F2, A2,B1))[1]
+end
+
+@testset "Presentation" begin
+
+	# over Integers
+	#=R, (x,y,z) = PolynomialRing(ZZ, ["x", "y", "z"])
+	generator_matrices = [R[x x^2*y; y y^2*x^2], R[x x^2*y; y y^2*x^2], R[x y; x^2*y y^2*x], R[x+R(1) x^10; x^2+y^2 y^4-x^4], R[42*x*y 7*x^2; 6*x 9*y^2]]
+	relation_matrices  = [R[x^3 y^4], R[x^3 y^4; x^2*y x*y^2], R[x*y^2 x; y^3 x*y^3], R[x x*y], R[3*x*y 7*x*y; 42 7]]
+	true_pres_matrices = [R[x^5*y-y^4 -x^5+x*y^3], R[-x^2*y-x*y-y x^2+x; x*y^4-x^3*y+y^4-x^2*y -x*y^3+x^3; -y^4+x^2*y x*y^3-x^3; x^2*y^3+x*y^3+y^3 -x^2*y^2-x*y^2], R[-x*y R(1); -x^2*y^5+x*y^3 R(0)], R[x^5-x*y^4+x^3*y+x*y^3 x^11-x^2*y-x*y; -x^5*y^7+x*y^11+2*x^5*y^6-x^3*y^8-3*x*y^10+2*x^5*y^5+2*x^3*y^7-3*x^5*y^4+2*x^3*y^6+5*x*y^8+2*x^5*y^3-3*x^3*y^5-5*x*y^7+2*x^3*y^4+2*x*y^6 -x^11*y^7+2*x^11*y^6+2*x^11*y^5-3*x^11*y^4+2*x^11*y^3+x^2*y^8-2*x^2*y^7+x*y^8-2*x^2*y^6-2*x*y^7+3*x^2*y^5-2*x*y^6-2*x^2*y^4+3*x*y^5-2*x*y^4], R[-13*y R(0); -377 -2639*x; -39 -273*x; -13*y-39 -273*x; -13 -91*x; y^2-42*x -294*x^2+21*x*y; 9*y^2-x+26*y+78 -7*x^2+189*x*y+546*x; -y^2+3*x 21*x^2-21*x*y]]
+	for (A,B,true_pres_mat) in zip(generator_matrices, relation_matrices, true_pres_matrices)
+    println(A,B,true_pres_mat)
+		SQ = Oscar.SubQuo(A,B)
+		pres_mat = matrix(Oscar.present_as_cokernel(SQ).quo)
+		@test Oscar.cokernel(pres_mat) == Oscar.cokernel(true_pres_mat)
+
+		pres_SQ, i = Oscar.present_as_cokernel(SQ, :both)
+		p = i.inverse_isomorphism
+		@test Oscar.iswelldefined(i)
+		@test Oscar.iswelldefined(p)
+		@test Oscar.isbijective(i)
+		@test Oscar.isbijective(p)
+	end=#
+
+
+
+
+	# over Rationals
+	R, (x,y,z) = PolynomialRing(QQ, ["x", "y", "z"])
+	generator_matrices = [R[x x^2*y; y y^2*x^2], R[x x^2*y; y y^2*x^2], R[x y; x^2*y y^2*x], R[x+R(1) x^10; x^2+y^2 y^4-x^4], R[x+R(1) x^10; x^2+y^2 y^4-x^4]]
+	relation_matrices  = [R[x^3 y^4], R[x^3 y^4; x^2*y x*y^2], R[x*y^2 x; y^3 x*y^3], R[x+y x+y; x*y x*y], R[x+y x+y; y x; x y]]
+	true_pres_matrices = [R[x^5*y-y^4 -x^5+x*y^3], R[-x^2*y-x*y-y x^2+x; -x^2*y^4+x^4*y x^2*y^3-x^4], R[-x*y R(1); -x^2*y^5+x*y^3 R(0)], R[-x^4+y^4-x^2-y^2 -x^10+x+R(1)], R[R(0) -x^2+y^2; -2*x^2 -x^9*y+x+1; -2*x*y^2 -x^8*y^3+y^2+x; -2*x*y^2+2*y^2 -x^8*y^3+x^7*y^3+y^2-1]]
+	for (A,B,true_pres_mat) in zip(generator_matrices, relation_matrices, true_pres_matrices)
+		SQ = Oscar.SubQuo(A,B)
+		pres_mat = matrix(Oscar.present_as_cokernel(SQ).quo)
+		F = Oscar.FreeMod(R,ncols(pres_mat))
+		@test Oscar.cokernel(F,pres_mat) == Oscar.cokernel(F,true_pres_mat)
+
+		pres_SQ, i = Oscar.present_as_cokernel(SQ, :both)
+    	p = i.inverse_isomorphism
+		@test Oscar.iswelldefined(i)
+		@test Oscar.iswelldefined(p)
+		@test Oscar.isbijective(i)
+		@test Oscar.isbijective(p)
+	end
+end
+
+
+#=@testset "Test kernel" begin
+	# This test doesn't terminate due to a bug in syz
```

Comment:
Is this bug in `syz` still present? If yes please insert a link to the relevant bug report(s). If no, we can activate this now?